keras_nlp/models/deberta/deberta_backbone.py

```diff
@@ -0,0 +1,274 @@
+                "kernel_initializer": keras.initializers.serialize(
+                    self.kernel_initializer
+                ),
+            }
+        )
+        return config
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class DebertaBackbone(keras.Model):
+    """DeBERTa encoder network.
+
+    This network implements a bi-directional Transformer-based encoder as
+    described in
+    ["DeBERTaV3: Improving DeBERTa using ELECTRA-Style Pre-Training with Gradient-Disentangled Embedding Sharing"](https://arxiv.org/abs/2111.09543).
+    It includes the embedding lookups and transformer layers, but does not
+    include the enhanced masked decoding head used during pretraining.
+
+    The default constructor gives a fully customizable, randomly initialized
+    DeBERTa encoder with any number of layers, heads, and embedding
+    dimensions. To load preset architectures and weights, use the `from_presets`
+    constructor.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind.
+
+    Args:
+        vocabulary_size: int. The size of the token vocabulary.
+        num_layers: int. The number of transformer layers.
+        num_heads: int. The number of attention heads for each transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: int. The size of the transformer encoding layer.
+        intermediate_dim: int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each transformer.
+        dropout: float, defaults to 0.1. Dropout probability for the
+            DeBERTa model.
+        max_sequence_length: int, defaults to 512. The maximum sequence length
+            this encoder can consume. The sequence length of the input must be
+            less than `max_sequence_length`.
+        bucket_size: int, defaults to 256. The size of the relative position
+            buckets. Generally equal to `max_sequence_length // 2`.
+
+    Example usage:
+    ```python
+    input_data = {
+        "token_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=128100),
+        "padding_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+
+    # Randomly initialized DeBERTa model
+    model = keras_nlp.models.DebertaBackbone(
+        vocabulary_size=128100,
+        num_layers=12,
+        num_heads=6,
+        hidden_dim=384,
+        intermediate_dim=1536,
+        max_sequence_length=512,
+        bucket_size=256,
+    )
+
+    # Call the model on the input data.
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.1,
+        max_sequence_length=512,
+        bucket_size=256,
```

Comment:
What happens if bucket_size is not max_sequence_length / 2. Would things blow up?

If so, we shouldn't offer a false configuration option here.

Reply:
Nope, it won't break. I've tried different `bucket_sizes`.